diff -urN mysqlcc-0.9.4-src.org/shared/src/CMySQL.cpp mysqlcc-0.9.4-src/shared/src/CMySQL.cpp
--- mysqlcc-0.9.4-src.org/shared/src/CMySQL.cpp	2003-12-14 00:10:44.000000000 +0100
+++ mysqlcc-0.9.4-src/shared/src/CMySQL.cpp	2005-04-11 23:30:11.000000000 +0200
@@ -224,9 +224,12 @@
   else
   {
     isconnected = true;
-    textcodec = QTextCodec::codecForName(mysqlGetCharsetName(), 2);
     initServerVersion();
+    QString charset = mysqlGetCharsetName();
+    textcodec = QTextCodec::codecForName(charset, 2);
     emit connected();
+    sprintf(init_command, "SET NAMES '%s'", charset.latin1());
+    mysql_query(mysql, init_command);
   }
   
   return isconnected;
@@ -566,7 +569,9 @@
 #ifdef DEBUG
   qDebug("CMySQL::mysqlGetCharsetName()");
 #endif
-  
+
+  if (version().major >= 4 && version().minor >= 0)
+    return  "utf8";
   if (isConnected())
     return  mysql_character_set_name(mysql);
   else
