--- mysqlcc-0.9.4-src/src/CApplication.cpp.orig	2003-12-14 00:10:43.000000000 +0100
+++ mysqlcc-0.9.4-src/src/CApplication.cpp	2004-02-01 18:45:15.865393376 +0100
@@ -109,8 +109,8 @@
   mydir += "/";
 #else
   //Directory where mysqlcc.exe (or mysqlcc) is.
-  QString mydir = QDir::currentDirPath() + "/";
-  //QString mydir = "/usr/share/mysqlcc/";  This needs to be done with a PREFIX define ..
+  //QString mydir = QDir::currentDirPath() + "/";
+  QString mydir = "/usr/share/mysqlcc/";  //This needs to be done with a PREFIX define ..
 #endif
 
   //Translations path
